motionplan/motionPlanner.go

```diff
@@ -385,3 +382,92 @@ IK:
 	}
 	return orderedSolutions, nil
 }
+
+// CheckPlan checks if obstacles intersect the trajectory of the frame following the plan.
+func CheckPlan(
+	checkFrame frame.Frame,
+	plan []map[string][]frame.Input,
+	startingConfiguration map[string][]frame.Input,
```

Comment:
@biotinker is this in line with what you had in mind when you said 

> we'll also need to pass in the current nominal position of the robot along the plan

Reply:
No; this needs to be a pose, not inputs.